File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/xmi/AttributesVisitor.java
package org.kevoree.modeling.api.xmi;

import org.kevoree.modeling.api.ModelAttributeVisitor;
import org.kevoree.modeling.api.meta.MetaAttribute;

class AttributesVisitor implements ModelAttributeVisitor {

    private SerializationContext context;

    public AttributesVisitor(SerializationContext context) {
        this.context = context;
    }

    public void visit(MetaAttribute metaAttribute, Object value) {
        if (value != null) {
            if (context.ignoreGeneratedID && metaAttribute.metaName().equals("generated_KMF_ID")) {
                return;
            }
            context.printer.append(" " + metaAttribute.metaName() + "=\"");
            XMIModelSerializer.escapeXml(context.printer, value.toString());
            context.printer.append("\"");
        }
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/xmi/AttributesVisitor.ts
///<reference path="../ModelAttributeVisitor.ts"/>
///<reference path="../meta/MetaAttribute.ts"/>

class AttributesVisitor implements ModelAttributeVisitor {

  private context: SerializationContext = null;

  constructor(context: SerializationContext) {
    this.context = context;
  }

  public visit(metaAttribute: MetaAttribute, value: any): void {
    if (value != null) {
      if (this.context.ignoreGeneratedID && metaAttribute.metaName().equals("generated_KMF_ID")) {
        return;
      }
      this.context.printer.append(" " + metaAttribute.metaName() + "=\"");
      XMIModelSerializer.escapeXml(this.context.printer, value.toString());
      this.context.printer.append("\"");
    }
  }

}

